lib/everest.cpp

```diff
@@ -20,10 +20,10 @@ namespace Everest {
 const auto remote_cmd_res_timeout_seconds = 300;
 
 Everest::Everest(std::string module_id, Config config, bool validate_data_with_schema,
-                 const std::string& mqtt_server_address, const std::string& mqtt_server_port,
-                 const std::string& mqtt_everest_prefix, const std::string& mqtt_external_prefix) :
-    mqtt_abstraction(MQTTAbstraction::get_instance(mqtt_server_address, mqtt_server_port, mqtt_everest_prefix,
-                                                   mqtt_external_prefix)),
+                 const std::string& mqtt_server_address, int mqtt_server_port, const std::string& mqtt_everest_prefix,
```

Comment:
couldn't we also use mqtt_broker_host and mqtt_broker_port here?

Reply:
absolutely, I'll change that in an upcoming commit

Reply:
I would leave this for now and tackle this in an upcoming PR with more cleanup